src/Publish/Microsoft.NETCore.Sdk.Publish.Tasks/Tasks/TransformWebConfig.cs

```diff
@@ -74,6 +82,11 @@ public override bool Execute()
             string outputFile = Path.GetFileName(TargetPath);
             XDocument transformedConfig = WebConfigTransform.Transform(webConfigXml, outputFile, IsAzure, IsPortable);
 
+            if (string.IsNullOrEmpty(ProjectGuid) && !IgnoreProjectGuid)
+            {
+                ProjectGuid = WebConfigTransform.GetProjectGuidFromSolutionFile(SolutionPath, ProjectFullPath);
+            }
+
             // Add the projectGuid to web.config if it is not present.
             transformedConfig = WebConfigTransform.AddProjectGuidToWebConfig(transformedConfig, ProjectGuid, IgnoreProjectGuid);
```

Comment:
> when the IgnoreProjectGuid is set

... manually?

So it seems that devs just not wanting to see this in their file should set it within their project properties. I sort'a kind'a wish that `IgnoreProjectGuid` was set :sparkles: magically :sparkles: to `true` when the opt out env var is set.